ssr/src/page/token/info.rs

```diff
@@ -131,7 +151,12 @@ pub fn TokenInfo() -> impl IntoView {
 
         let cans = unauth_canisters();
         let meta = token_metadata_by_root(&cans, params.user_principal, params.token_root).await?;
-        Ok(meta.map(|m| (m, (params.token_root, params.user_principal))))
+        let root = cans.sns_root(params.token_root).await;
+        let sns_cans = root.list_sns_canisters(ListSnsCanistersArg {}).await?;
+        let Some(ledger) = sns_cans.ledger else {
+            return Err(ServerFnError::new("Couldn't find ledger"));
+        };
+        Ok(meta.map(|m| (m, (params.token_root, params.user_principal), ledger)))
```

Comment:
You can add a new field for the ledger principal here: https://github.com/yral-dapp/hot-or-not-web-leptos-ssr/blob/0bddfeb068acedb52fbe3d5b3502ea417b7e78b8/ssr/src/utils/token/mod.rs#L249